src/test/java/com/kodilla/ecommercee/domain/ProductTestSuite.java

```diff
@@ -0,0 +1,165 @@
+package com.kodilla.ecommercee.domain;
+
+import com.kodilla.ecommercee.repository.GroupRepository;
+import com.kodilla.ecommercee.repository.ProductRepository;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.math.BigDecimal;
+import java.util.List;
+import java.util.Optional;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest
+public class ProductTestSuite {
+    @Autowired
+    private ProductRepository productRepository;
+    @Autowired
+    private GroupRepository groupRepository;
+
+    @Transactional
+    @Test
+    public void testSaveProduct() {
+        //given
+        Product testProduct = new Product();
+        testProduct.setName("Shirt");
+        testProduct.setDescription("white shirt");
+        testProduct.setPrice(BigDecimal.valueOf(5.20));
+
+        //when
+        productRepository.save(testProduct);
+        List<Product> allProducts = productRepository.findAll();
+
+        //then
+        Assert.assertTrue(allProducts.contains(testProduct));
```

Comment:
W mojej ocenie przy obecnej konfiguracji H2 transakcja jest zwijana po każdym teście - o ile jest opatrzona @Transactional. Więc można było normalnie testować ilości... poprawcie mnie proszę jeśli się mylę bo jest już późno :)

2019-02-14 00:19:52.985  INFO 20154 --- [           main] o.s.t.c.transaction.TransactionContext   : Rolled back transaction for test:

Reply:
Teoretycznie tak powinno być, ale skoro zaczęliśmy już pisać w jeden, "mega bezpieczny" sposób, dlatego tak też kontynuowałem

Reply:
no i właśnie tu pojawia się problem. Bo w mojej ocenie polegamy na tym, że inne testy będą napisane w ten "bezpieczny" sposób. 

OK, zostawmy.